src/debug/panic.c

```diff
@@ -88,5 +88,9 @@ void __panic(uint32_t p, char *filename, uint32_t linenum)
 				 filename, strlen + 1));
 	}
 
+	/* To distinguish regular panic() calls from exceptions */
+	__asm__ __volatile__("movi a3, 0\n\twsr.exccause a3\n\tesync" : : :
```

Comment:
1. It looks much better if every instruction is in separate line.
2. 0 is a valid exception code as you mentioned, so I wouldn't use it. You can define exccause maximum value (63) and use it here instead. In the same commit you can add the same value to the coredumper and describe as artificially caused exception.

Reply:
Addressed (I think)